### PR TITLE
Update engine api for snapshot parameters and rewrite params default to none when json is empty

### PR DESCRIPTION
**Pull request by nhlien93**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build was run locally and any changes were pushed
- [x] Lint has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current vs new behavior?
There are a couple things in this change:
1) The engine api is changed to 1.12.0. This is necessary to get our trunk delphix engine to work, after the delphix engine changes get backported to an older api we will have to make another change
2) The snapshot parameters definition schema is an actual object rather than just a json v4. therefore the build has to be updated.
3) Swagger generated objects are empty if the json blob is None. I change the behavior to make the object None as well.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: platform/src/main/python/dlpx/virtualization/platform/_linked.py [hunk 1]
```diff
@@ -142,8 +142,10 @@ def _internal_direct_pre_snapshot(self, request):
             json.loads(request.repository.parameters.json))
         source_config = SourceConfigDefinition.from_dict(
             json.loads(request.source_config.parameters.json))
-        snapshot_parameters = SnapshotParametersDefinition.from_dict(
-            json.loads(request.snapshot_parameters.parameters.json))
+        snap_params = json.loads(request.snapshot_parameters.parameters.json)
+        snapshot_parameters = (
+            None if snap_params is None else
+            SnapshotParametersDefinition.from_dict(snap_params))
```
Comment (ankursarin) [inline platform/src/main/python/dlpx/virtualization/platform/_linked.py:148]: nit: leave a comment about why we do this?

**Review by ankursarin — APPROVED** [review]
looks good to me

File: platform/src/main/python/dlpx/virtualization/platform/_linked.py [hunk 2]
```diff
@@ -142,8 +142,14 @@ def _internal_direct_pre_snapshot(self, request):
             json.loads(request.repository.parameters.json))
         source_config = SourceConfigDefinition.from_dict(
             json.loads(request.source_config.parameters.json))
-        snapshot_parameters = SnapshotParametersDefinition.from_dict(
-            json.loads(request.snapshot_parameters.parameters.json))
+        snap_params = json.loads(request.snapshot_parameters.parameters.json)
+        #
+        # The snapshot_parameters object should be set to None of the json from
```
Comment (jeffngo) [inline platform/src/main/python/dlpx/virtualization/platform/_linked.py:147]: Here and everywhere else: spelling nit: of -> if

**Review by jeffngo — APPROVED** [review]

**Review by ravi-cm — APPROVED** [review]
Looks good.

**Pull request merged**